Rakefile

```diff
@@ -71,6 +75,63 @@ def re_run_basic_smoke_test
   sh beaker
 end
 
+def get_passing_json(current_branch_trunk)
+  jenkins_url = "https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/#{current_branch_trunk}/view/Suite/job/platform_puppet-agent_intn-van-promote_suite-daily-promotion-#{current_branch_trunk}/lastSuccessfulBuild/api/json"
+  @memoized_passing_json ||= Net::HTTP.get(URI.parse(jenkins_url))
+end
+def get_passing_puppetagent_sha(current_branch_trunk)
+  get_passing_json(current_branch_trunk)
+  JSON.parse(@memoized_passing_json)['actions'][0]['parameters'].find{|x| x['name'] == 'SUITE_COMMIT'}['value']
+end
+def get_passing_puppet_sha(current_branch_trunk)
+  get_passing_json(current_branch_trunk)
+  JSON.parse(@memoized_passing_json)['actions'][0]['parameters'].find{|x| x['name'] == 'puppet_COMPONENT_COMMIT'}['value']
+end
+
+PUPPET_SUBMODULE_PATH = File.join('ruby','puppet')
+def get_passing_puppet_version(current_branch_trunk)
+   #FIXME: this should be updated when the package yaml file contains the metadata we need
+   #  we have to replace the hyphens with dots because, vanagon
+  `cd #{PUPPET_SUBMODULE_PATH}; git describe`.strip.gsub(/-/,'.')
+end
+
+def replace_puppet_pins(passing_puppetagent_sha, passing_puppet_version)
+  # read beaker options hash from its file
+  beaker_options_file_path = 'acceptance/config/beaker/options.rb'
+  puts("replacing puppet sha and version in #{beaker_options_file_path}, with agent sha: #{get_passing_puppetagent_sha(current_branch_trunk)} and puppet version: #{get_passing_puppet_version(current_branch_trunk).to_s}")
```

Comment:
I think we could use the passed in values here instead of calling to the accessors methods and returning their memoized values.